integrations/google-analytics-4/lib/index.js

```diff
@@ -0,0 +1,275 @@
+     * https://developers.google.com/analytics/devguides/collection/ga4#add_an_additional_google_analytics_property_to_an_existing_tag
+     */
+    for (var i = 0; i < measurementIds.length; i++) {
+      window.gtag('config', measurementIds[i], config)
+
+    }
+
+    /**
+     * Set persistent values shared across all gtag.js usage.
+     * https://developers.google.com/gtagjs/reference/api#set
+     */
+    for (var i = 0; i < sets.length; i++) {
+      window.gtag.apply(null, sets[i]);
+    }
+
+    self.ready();
+  });
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+GA4.prototype.loaded = function() {
+  return !!(
+    window.ga4DataLayer && Array.prototype.push !== window.ga4DataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Facade.Identify} event
+ */
+GA4.prototype.identify = function(identify) {
+  var opts = this.options;
+  var userPropertyMappings = opts.userProperties;
+
+  var userProperties = {};
+
+  // Map all customer-defined user property mappings.
+  for (var eventField in userPropertyMappings) {
+    if (!userPropertyMappings.hasOwnProperty(eventField)) {
+      continue;
+    }
+
+    var userProp = userPropertyMappings[eventField];
+    var value = identify.proxy(eventField);
+
+    userProperties[userProp] = value;
+  }
+
+    /**
+     * Map the user_id property the Send User ID setting is enabled. Note that the user ID
+     * can be appended as part of the user_properties object instead of being configured by
+     * an explicit command.
+     * https://developers.google.com/analytics/devguides/collection/ga4/cookies-user-id#set_user_id
+     */
+  var userId = identify.userId();
+  if (opts.sendUserId && userId) {
+    userProperties.user_id = userId;
+  }
+
+  if (Object.keys(userProperties).length) {
+    window.gtag('set', 'user_properties', userProperties);
+  }
+};
+
+/**
+ * Group
+ *
+ * @api public
+ * @param {Facade.Group} group
+ */
+GA4.prototype.group = function(group) {
+  window.gtag('event', 'join_group', {
+    group_id: group.groupId()
+  });
+};
+
+/**
+ * Page
+ *
+ * @api public
+ * @param {Facade.Page} page
+ */
+GA4.prototype.page = function(page) {
+  // If the Disable Page View Measurement setting is set to false then
+  // don't handle page calls to avoid duplicate page_view events.
+  if (!this.options.disablePageViewMeasurement) {
+    return;
+  }
+
+  // Track categorized pages setting needed here?
+
+  var props = page.properties();
+  var name = page.fullName();
+
+  var pageLocation = props.url;
+  var pageReferrer = page.referrer();
+  var pageTitle = name || props.title;
+
+  window.gtag('event', 'page_view', {
+    page_location: pageLocation,
+    page_referrer: pageReferrer,
+    page_title: pageTitle
+  });
+};
+
+/**
+ * Track
+ *
+ * @api public
+ * @param {Track} track
+ */
+
+GA4.prototype.track = function(track) {
+
+  var mappings = this.options.customEventsAndParameters;
+
+  for (var i = 0; i < mappings.length; i++) {
```

Comment:
Small nit: Sometimes it is hard to comprehend the Mixed Setting type. I know for AA we include a code comment re: format( [seen here](https://github.com/segmentio/analytics.js-integrations/blob/ede07d097a7b447a58e2f08cbb2c0eefbaf828cb/integrations/adobe-analytics/lib/index.js#L712-L730) ). Might be helpful to include one here for other contributors less familiar with this settings ie. 
```
{
  segmentEvent: 'stringSegmentEvent',
  googleEvent: 'stringGoogleEvent',
  parameters: { segmentParam: 'googleParam'}
}
```

Reply:
Not sure if ☝️ is even the right format, based off below looks like parameters is an array of objects?

Reply:
Good point. I'll add an example 🙌🏽